packages/reactor/docs/planning/Jobs/index.md

```diff
@@ -0,0 +1,116 @@
+
+### Signature Verification
+
+In all cases, the `IJobExecutor` will verify that the `Action` signatures are valid.
+
+If the `Action` signature is not valid, the `IJobExecutor` will emit a `jobFailed` event and the job will not be retried.
+
+### Operation Verification 
+
+Optionally, a `Job` may contain information about expected `Operation` data. In this case, the executor will verify that resulting `Operation` hash matches the expected `Operation` hash.
+
+This is useful for the synchronization flow, where we receive `operation` objects over a network and need to apply them locally.
+
+### Error Handling
+
+The executor emits a set of structured events so that clients can react to job progress and failures:
+
+- **`jobStarted`** - issued when execution of a job begins.
+- **`jobCompleted`** - issued after a job finishes successfully.
+- **`jobRetry`** - issued when execution throws an error and the executor will retry the job.
+- **`jobFailed`** - issued when execution throws an error and will not be retried.
+
+#### Retry Logic
+
+Retries will only be attempted if the job failed for a reason that is likely to be resolved by retrying. For example, if the job failed because the operation was already applied or because the operation was invalid, it will not be retried.
+
+In some cases, a retry might result in a requeue instead, where the job is added back to the queue to wait for some other job to be processed.
+
+If a job failed because of a temporary network issue, for example, it will be retried with exponential backoff + jitter.
+
+#### Fatal Errors
+
+If a `jobFailed` event is fired, we know that a set of operations could not be applied. We then follow a set of steps to try to resolve the issue:
+
+1. Sync missing history – Attempt to finishin syncing from any remotes, for the documents affected, before continuing.
+
+2. Re‑apply deterministically – Once caught up, retry applying the operation.
+
+3. Bubble up a sync failure if still inconsistent – If the operation still cannot be applied even after resyncing, the reactor itself will emit a failure event (`fatalError`) for monitoring but will not persist the failure in the operations store. The failing reactor can mark the operation for manual reconciliation or trigger a rollback to some previous state.
+
+### Dependencies
+
+- [IQueue](../Queue/index.md)
+- [IEventBus](../Events/index.md)
+- [IOperationStore](../Reactor/Interfaces/IOperationStore.md)
+
+### Links
+
+- [Interface](interface.md)
+- [Usage](usage.md)
+
+### Notes
+
+- The job executor listens for 'jobAvailable' events from the event bus instead of polling the queue.
+- When a 'jobAvailable' event is received, the executor checks if it has capacity to process more jobs.
+- If capacity allows, the executor pulls a job from the queue and executes it.
+- When a job is fully executed, it tries to pull another job from the queue.
+- Jobs should be executed in the correct order per document/scope/branch to maintain consistency.
```

Comment:
I think we need to introduce cross-document/scope/branch dependencies.
For example, we want the first operation to be run for a document to be the `ACTION_CREATE`, so operations on other scopes are blocked by it. The same for version upgrades.

Will linked `authorization` scopes also create dependencies?

Reply:
That's a good call-out. I don't have a plan there other than "the queue worries about that".